tests/unit/test_rbac_resolvers_key_value_pair.py

```diff
@@ -17,39 +17,433 @@
-        permission_types = PermissionType.get_valid_permissions_for_resource_type(
-            ResourceType.KEY_VALUE_PAIR)
+        grant_db = PermissionGrantDB(
+            resource_uid=self.resources["system_role_4"].get_uid(),
+            resource_type=ResourceType.KEY_VALUE_PAIR,
+            permission_types=[PermissionType.KEY_VALUE_DELETE],
+        )
+        grant_db = PermissionGrant.add_or_update(grant_db)
+        permission_grants = [str(grant_db.id)]
+        role_5_db = RoleDB(
+            name="custom_role_key_value_pair_delete_grant",
+            permission_grants=permission_grants,
+        )
+        role_5_db = Role.add_or_update(role_5_db)
+        self.roles["custom_role_key_value_pair_delete_grant"] = role_5_db
+
+        # Create some mock role assignments
+        user_db = self.users["custom_role_key_value_pair_view_grant"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["custom_role_key_value_pair_view_grant"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["custom_role_key_value_pair_list_grant"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["custom_role_key_value_pair_list_grant"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["custom_role_key_value_pair_all_grant"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["custom_role_key_value_pair_all_grant"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["custom_role_key_value_pair_create_grant"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["custom_role_key_value_pair_create_grant"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["custom_role_key_value_pair_delete_grant"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["custom_role_key_value_pair_delete_grant"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["st2kv_system_grant"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["st2kv_system_grant"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+    def test_user_resource_db_permissions(self):
```

Comment:
This test covers too much. It's also missing some test cases. 

Can you separate this out to multiple tests? Please systematically refactor the unit tests according to the following list.

- System roles
  - System Observer role on all system scoped KVPs
    - Succeed on LIST permission
    - Succeed on VIEW permission
    - Fail on SET permission
    - Fail on DELETE permission
  - System Admin role on all system scoped KVPs
    - Succeed on ALL permission
    
- Default for user on user owned KVPs
  - No specific roles and grants need to be defined for user to set/delete their own KVPs.
  - User should be defined access on ALL permission to another user's KVPs.
  - Succeed on ALL permission to user owned KVPs

- User roles with permission on specific system scope KVPs
  - Custom user role with LIST on specific system scope KVP
    - Succeed on LIST permission
    - Should VIEW permission succeed or fail here?
    - Fail on SET permission
    - Fail on DELETE permission
  - Custom user role with VIEW on specific system scope KVPs
    - Should LIST permission succeed or fail here?
    - Succeed on VIEW permission
    - Fail on SET permission
    - Fail on DELETE permission
  - Custom user role with SET on specific system scope KVPs
    - Should LIST permission succeed or fail here?
    - Succeed on VIEW permission
    - Succeed on SET permission
    - Fail on DELETE permission
  - Custom user role with DELETE on specific system scope KVPs
    - Should LIST permission succeed or fail here?
    - Succeed on VIEW permission
    - Succeed on SET permission
    - Succeed on DELETE permission

Reply:
@ashwini-orchestral Please review the outline above. I highlighted the unit tests for each of the test cases in the outline here.